File: MelissaHong/Algorithms/basic13.js
//This took me about an hour to complete

//Print 1 to 255 integers
function get_array(){
var arr=[];
for (var i=1; i<=255; i++){
	arr.push(i);
}
return arr;
}

//Print Odds 1 to 255
function get_odds(){
	var arr=[];
	for (var i=1; i<=255; i++){
		if (i%2===1){
			arr.push(i);
		}
	}
	return arr;
}

//PrintIntsAndSum0to255()
function intsandsum(){
	var arr=[];
	var sum=0;
	for (var i=0; i<=255; i++){
			arr.push(i);
			sum+=i;
			console.log(sum);
		}
	console.log(arr);
	}

//Print Array Vals
function vals(){
	var arr=[1,2,3,4];
	console.log(arr);
}

//Print Max Array
function findMax(){
	var max=0;
	var arr=[1,2,3,4];
	for (var i=0; i<arr.length; i++){
		if (arr[i]>max){
			max=arr[i];
		}
	}
	console.log(max);
}

//Print Avg Array
function findAvg(){
	var arr=[10,20,30,40];
	var sum=0;
	for (var i=0; i<arr.length; i++){
		sum+=arr[i];
	}
	var avg=sum/arr.length;
	console.log(avg);
}

//Return Odds 1 to 255
function returnOdds(){
	var arr=[];
	for (var i=1; i<=255; i++){
		if(i%2===1){
		arr.push(i);
		}
	}
	console.log(arr);
}

//Square value of array #s
function squareArray(){
	var arr=[1,2,3,4];
	for (var i=0; i<arr.length; i++){
		arr[i]=arr[i]*arr[i];
	}
	return(arr);
}

//Return array count greater than Y (arr, y)
function greaterThanY(){
	var Y=0;
	var arr=[1,2,3,4];
	var count=0;
	for (var i=0; i<arr.length; i++){
		if (arr[i]>Y){
			count++
		}
	}
	return count;
}

//Set neg values to zero of an array
function zeroNeg(){
	var arr=[0, -2, -3, -5, 1];
	for (var i=0; i<arr.length; i++){
		if (arr[i]<0){
			arr[i]=0;
		}
	}
	return arr;
}

//Max Min Avg values
function maxMinAvg(){
	var arr=[0,1,2,3,4];
	var max=arr[0];
	var min=arr[0];
	var sum=arr[0];
	for (var i=0; i<arr.length;i++){
		if (arr[i]>max){
			max=arr[i];
		}
		if (arr[i]<min){
			min=arr[i];
		}
		sum+=arr[i];
		var avg=sum/arr.length;
	}
	var maxminavg=[max, min, avg];
	return maxminavg;
}

//Shift the array by 1, with index 0 at the end

function shiftOne(arr){
	var arr=[0,2,4,6,8];
	for (var i=0; i<arr.length; i++){
		arr[i]=arr[i+1];
	}
	arr[4]=0;
	}
	return arr;
}


//Replace neg values to dojo
function negDojo(){
	var  arr=[0, -2, -6, 8];
	for (var i=0; i<arr.length; i++){
		if (arr[i]<0){
			arr[i]='dojo';
		}
	}
	return arr;
}

File: MelissaHong/Algorithms/basic13.py
def get_array():
    array = []
    for i in range (1,256):
        array.append(i)
    print array

get_array()

def get_odds():
    array = []
    for i in range (1,256):
        if i%2==1:
            array.append(i)
    print array

get_odds()

def intsandsum():
    array = []
    sum = 0
    for i in range (0,256):
        array.append(i)
        sum += i
        print sum
    print array

intsandsum()

arr = [2,4,12,44]
def vals(values_list):
    for i in range (len(arr)):
        print "array " + str(i) + " is equal to " + str(arr[i])

vals(arr)

#def findMax(arr):
arr = [3,12,5,22,1]

def findMax(arr):
    max = 0
    for i in range (len(arr)):
        if arr[i] > max:
            max = arr[i]
    print max

findMax(arr)

arr = [3,12, 55, 1, 3]
def findAvg(arr):
    sum = 0
    for i in range (len(arr)):
        sum += arr[i]
        avg = sum/len(arr)
    print avg

findAvg(arr)

def returnOdds():
    arr = []
    for i in range (1, 256):
        if i%2==1:
            arr.append(i)
    print arr

returnOdds()


def squareArray():
    for i in range (len(arr)):
        arr[i] = arr[i] * arr[i]
    print arr

squareArray()

arr = [4,1,12,0]
Y = 3
def greaterThanY(arr, Y):
    Y = 0
    count = 0
    for i in range (len(arr)):
        if (arr[i] > Y):
            count += 1
    print count

greaterThanY(arr, Y)

arr = [-1,-2,0,1,2]
def zeroNeg(arr):
    for i in range (len(arr)):
        if arr[i] < 0:
            arr[i] = 0
    print arr

zeroNeg(arr)

arr = [1,2,3,0,2,12]
def maxMinAvg(arr):
    max = arr[0]
    min = arr[0]
    sum = arr[0]
    for i in range (len(arr)):
        if arr[i] > max:
            max = arr[i]
        if arr[i] < min:
            min = arr[i]
        sum += arr[i]
        avg = sum/len(arr)
    maxminavg = [max, min, avg]
    print maxminavg

maxMinAvg(arr)

arr = [1,2,3,4]
temp = arr[0]
def shiftOne(arr):
    for i in range(1, len(arr)):
        arr[i-1] = arr[i]
    arr[len(arr)-1] = temp
    print arr

shiftOne(arr)
